pages/learn/miscellaneous-examples.md

```diff
@@ -157,3 +173,291 @@ We also introduce the following with this example:
+
+```json
+{
+  "$id": "https://example.com/regex-pattern.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Regular Expression Pattern",
+  "type": "object",
+  "properties": {
+    "code": {
+      "type": "string",
+      "pattern": "^[A-Z]{3}-\\d{3}$"
+    }
+  }
+}
+```
+
+**Data**
+
+```json
+{
+  "code": "ABC-123"
+}
+```
+
+The provided data, "ABC-123", satisfies this pattern defined in the schema.
+
+
+## Complex object with nested properties
+
+The schema below represents a complex object with various properties including `name`, `age`, `address`, and `hobbies`. The `address` property is an object with nested properties, and the `hobbies` property is an array of strings. The `name` and `age` properties are required.
+
+```json
+{
+  "$id": "https://example.com/complex-object.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Complex Object",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "age": {
+      "type": "integer",
+      "minimum": 0
+    },
+    "address": {
+      "type": "object",
+      "properties": {
+        "street": {
+          "type": "string"
+        },
+        "city": {
+          "type": "string"
+        },
+        "state": {
+          "type": "string"
+        },
+        "postalCode": {
+          "type": "string",
+          "pattern": "\\d{5}"
+        }
+      },
+      "required": ["street", "city", "state", "postalCode"]
+    },
+    "hobbies": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": ["name", "age"]
+}
+```
+
+**Data**
+
+```json
+{
+  "name": "John Doe",
+  "age": 25,
+  "address": {
+    "street": "123 Main St",
+    "city": "New York",
+    "state": "NY",
+    "postalCode": "10001"
+  },
+  "hobbies": ["reading", "running"]
+}
+```
+
+The provided data conforms to the schema by including values for the required properties and ensuring the `age` is an integer greater than or equal to zero. The `address` object contains all the necessary properties, and the `hobbies` property is an array of strings.
+
+
+## Conditional validation with dependentRequired
+
+In this example, the `dependentRequired` keyword is used to specify that the property `bar` is required when the property `foo` is present. The schema enforces the condition that if `foo` exists, then `bar` must also be present.
+
+```json
+{
+  "$id": "https://example.com/conditional-validation-dependentRequired.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Conditional Validation with dependentRequired",
+  "type": "object",
+  "properties": {
+    "foo": {
+      "type": "boolean"
+    },
+    "bar": {
+      "type": "string"
+    }
+  },
+  "dependentRequired": {
+    "foo": ["bar"]
+  }
+}
+```
+
+**Data**
+
+```json
```

Comment:
It might be good to show an invalid instance where `bar` is missing as well as a valid instance where both `foo` and `bar` are missing, and explanations why.